simplify samples/crop_models and README organization

itsjohnward opened the pull request:
Moved in https://github.com/gro-intelligence/api-client/pull/93 into its own directory. Expect to see it reorganized back up into the main crop_models folder in the future, but as a more immediate fix, point this link at the Brazil soybeans directory

itsjohnward committed:
Update Brazil soybeans notebook link in main README

Moved in https://github.com/gro-intelligence/api-client/pull/93 into its own directory. Expect to see it reorganized back up into the main crop_models folder in the future, but as a more immediate fix, point this link at the Brazil soybeans directory

nemozen commented on README.md:
````diff
@@ -47,7 +47,7 @@ If the API client is installed and your authentication token is set, a csv file
 python crop_models/soybeans.py
 ```
 
-3. See [brazil_soybeans.ipynb](https://github.com/gro-intelligence/api-client/blob/development/api/client/samples/crop_models/brazil_soybeans.ipynb) for a longer, more detailed demonstration of many of the API's capabilities in the form of a Jupyter notebook.
````
let's do the reverse as discussed

nemozen committed:
Revert "Update Brazil soybeans notebook link in main README"

This reverts commit f481a10033f3bc7310aaf1731cd2fc35c0b38b86.

nemozen committed:
Revert "Move notebook to own folder"

This reverts commit 32d907a16be6d60a5e286fdb68fca20ee7da054d.

nemozen committed:
single  requirements.txt for crop_models

nemozen committed:
merge requirements from sugar content model

nemozen committed:
move sugar content crop model to crop_models

JGrinst commented:
As discussed, I don't think this is the reorganization we want. We should not force requirements on models that do not need them. Brazil TRS model does not need geopandas and we shouldn't make users think it's a requirement, since windows users would then also need anaconda to run it.

If we want to have a single requirements doc for crop models, that's fine, but then Brazil TRS shouldn't be a crop model. Let's bring it up to it's own folder.

nemozen committed:
merged into parent dir

nemozen commented:
ok updated and tested both examples

nemozen commented:
> As discussed, I don't think this is the reorganization we want. We should not force requirements on models that do not need them. Brazil TRS model does not need geopandas and we shouldn't make users think it's a requirement, since windows users would then also need anaconda to run it.
> 
> If we want to have a single requirements doc for crop models, that's fine, but then Brazil TRS shouldn't be a crop model. Let's bring it up to it's own folder.


Yup... turns out the only diff is seaborn which is a simple viz package and statsmodel where we use one regression model which overlaps with sklearn. Neither one is particularly onerous but we can clean that up and reduce the deps a bit separately.

nemozen committed:
clean up README files

- in windows-setup, make conda recommendation more explicit
- fix: conda install pip was missing
- make readme apply to  crop_models in general
- don't repeat base instructions in the crop_models, link instead

nemozen committed:
whitespace cleanup

tolothy commented on api/client/samples/crop_models/requirements.txt:
```diff
@@ -1,7 +1,11 @@
-pandas
-matplotlib
+descartes
```
descartes is a dependency of shapely, which itself is a dependency of geopandas. does that mean we dont need to list it in requirements explicitly?

nemozen replied:
good question... I think it was added in the TRS notebook since that didn't need geopandas. Doesn't hurt to leave be more explicit here.

tolothy commented on api/client/samples/crop_models/requirements.txt:
```diff
@@ -1,7 +1,11 @@
+descartes
 geopandas
-scikit-learn
 ipython
 jupyter
```
do we only need one of the two?

nemozen replied:
ditto

nemozen commented:
> separately



> > As discussed, I don't think this is the reorganization we want. We should not force requirements on models that do not need them. Brazil TRS model does not need geopandas and we shouldn't make users think it's a requirement, since windows users would then also need anaconda to run it.
> > If we want to have a single requirements doc for crop models, that's fine, but then Brazil TRS shouldn't be a crop model. Let's bring it up to it's own folder.
> 
> Yup... turns out the only diff is seaborn which is a simple viz package and statsmodel where we use one regression model which overlaps with sklearn. Neither one is particularly onerous but we can clean that up and reduce the deps a bit separately.

We tested `conda install geopandas` from scratch on  windows and it worked.  As discussed, I have changed the main readme to say anaconda is recommended for windows.  

overall, we don't want to overwhelm the novice user with too many ifs in the readmes, we just want to pick the most pain free most general installation and recommend that. ( ditto for pip install vs git clone for example).

nemozen committed:
simplify README

nemozen commented:
> > separately
> 
> > > As discussed, I don't think this is the reorganization we want. We should not force requirements on models that do not need them. Brazil TRS model does not need geopandas and we shouldn't make users think it's a requirement, since windows users would then also need anaconda to run it.
> > > If we want to have a single requirements doc for crop models, that's fine, but then Brazil TRS shouldn't be a crop model. Let's bring it up to it's own folder.
> > 
> > 
> > Yup... turns out the only diff is seaborn which is a simple viz package and statsmodel where we use one regression model which overlaps with sklearn. Neither one is particularly onerous but we can clean that up and reduce the deps a bit separately.
> 
> We tested `conda install geopandas` from scratch on windows and it worked. As discussed, I have changed the main readme to say anaconda is recommended for windows.
> 
> overall, we don't want to overwhelm the novice user with too many ifs in the readmes, we just want to pick the most pain free most general installation and recommend that. ( ditto for pip install vs git clone for example).

cleaned up the same issue in ../prevented_plant/README.md  Now they all reference the common setup instructions, shouldbe much easier.   interestingly in that dir geopandas in windows is assumed to just work. ( But there's as separate issue with xgboost on windows, which is handled nicely in the code with a fallback alg).

itsjohnward commented on api/client/samples/crop_models/requirements.txt:
```diff
@@ -1,7 +1,11 @@
+pandas
+scikit-learn
+seaborn
+sklearn
```
Remove sklearn

On PyPI it exists as an old version https://pypi.org/project/sklearn/, but on Anaconda Cloud it doesn't exist at all and breaks the install. `scikit-learn` is the right one on both

nemozen replied:
done

itsjohnward commented on api/client/samples/crop_models/README.md:
```diff
@@ -0,0 +1,7 @@
+1. Follow the [instructions to install the Gro API Client](../../../../README.md)
+1. Download the [requirements file in this directory](./requirements.txt)
+2. Run `pip install -r requirements.txt` (for MacOS and Linux) or `conda install --file requirements.txt` (for Windows) to automatically install the required dependencies
+3. Download the notebook to your working directory
```
"notebook or script"? There are some python scripts mixed in here too - not all notebooks

And similarly on the next line, run jupyter notebook OR python script.py?

nemozen replied:
done

itsjohnward reviewed (commented):
Tested all the models from a fresh anaconda environment, and after removing sklearn from the requirements file they all work

nemozen committed:
Update README.md

Examples can be .ipynb or .py

nemozen committed:
Update requirements.txt

remove sklearn, scikit-learn supercedes it

nemozen committed:
Update README.md

and -> or

itsjohnward commented:
https://github.com/gro-intelligence/api-client/tree/e010553e35e37e330abbe8311107407c7f2f2b63/api/client/samples/crop_models

The brazil_soybeans folder is still there, with brazil_soybeans_model_overview.md

Want to remove that file and directory?

nemozen committed:
Delete brazil_soybeans_model_overview.md

actual notebook  moved to parent dir.  this  content is covered in the notebook itself and in the "api showcase" on the website.

nemozen commented:
> https://github.com/gro-intelligence/api-client/tree/e010553e35e37e330abbe8311107407c7f2f2b63/api/client/samples/crop_models
> 
> The brazil_soybeans folder is still there, with brazil_soybeans_model_overview.md
> 
> Want to remove that file and directory?

yup. 

> https://github.com/gro-intelligence/api-client/tree/e010553e35e37e330abbe8311107407c7f2f2b63/api/client/samples/crop_models
> 
> The brazil_soybeans folder is still there, with brazil_soybeans_model_overview.md
> 
> Want to remove that file and directory?

done

nemozen commented:
> "and" -> "or"

done

itsjohnward committed:
Update prevent_plant README numbering

nemozen committed:
Update windows-setup.md

  conda includes pip by default
